e2e/utilities/longhorn-install.sh

```diff
@@ -23,19 +25,46 @@ install_longhorn_by_chart(){
   wait_longhorn_status_running
 }
 
+install_longhorn_stable_by_chart(){
+  git clone --single-branch \
+            --branch "${LONGHORN_STABLE_VERSION}" \
+            "${LONGHORN_REPO_URI}" \
+            "${LONGHORN_REPO_DIR}"
+    helm upgrade --install longhorn "${LONGHORN_REPO_DIR}/chart/" --namespace "${LONGHORN_NAMESPACE}"
+    wait_longhorn_status_running
+}
+
+install_longhorn_stable_by_manifest(){
+  LONGHORN_STABLE_VERSION=${LONGHORN_STABLE_VERSION}
+  LONGHORN_STABLE_MANIFEST_URL="https://raw.githubusercontent.com/longhorn/longhorn/${LONGHORN_STABLE_VERSION}/deploy/longhorn.yaml"  
+  install_longhorn_by_manifest "${LONGHORN_STABLE_MANIFEST_URL}"
+}
+
 install_longhorn(){
   create_longhorn_namespace
   install_backupstores
   if [[ "${LONGHORN_INSTALL_METHOD}" == "helm" ]]; then
-    LONGHORN_REPO_URI=${LONGHORN_REPO_URI:-"https://github.com/longhorn/longhorn.git"}
-    LONGHORN_REPO_DIR="${TMPDIR}/longhorn"
     install_longhorn_by_chart
   elif [[ "${LONGHORN_INSTALL_METHOD}" == "manifest" ]]; then
     generate_longhorn_yaml_manifest "${TF_VAR_tf_workspace}"
     install_longhorn_by_manifest "${TF_VAR_tf_workspace}/longhorn.yaml"
   fi
   setup_longhorn_ui_nodeport
+}
 
+install_longhorn_stable_version(){
+  create_longhorn_namespace
+  install_backupstores
+  if [[ "${LONGHORN_INSTALL_METHOD}" == "helm" ]]; then
+    install_longhorn_stable_by_chart
+  elif [[ "${LONGHORN_INSTALL_METHOD}" == "manifest" ]]; then
+    install_longhorn_stable_by_manifest
+  fi
+  setup_longhorn_ui_nodeport
 }
 
-install_longhorn
+if [[ $# -gt 0 ]]; then
```

Comment:
Could you elaborate on why this is needed?

Reply:
@yangchiu, From file libs/longhorn_deploy/base.py of this PR, a new statement has been added to handle the following scenarios:
- longhorn-install.sh is run with no arguments, the `install_longhorn` function is called which is the default design as before.
- longhorn-install.sh is run with an argument, the function named by the argument is executed, in this test case, `install_longhorn_stable_version` will be executed to perform steps for keyword `Then Install Longhorn stable version`
```python
def install_longhorn(self, is_stable_version=False):
        current_path=os.getcwd()
        full_path = os.path.join(current_path, LONGHORN_INSTALL_SCRIPT_PATH)

        if is_stable_version is True:
            cmd = ['bash', full_path, LONGHORN_INSTALL_STABLE_SHELL_FUNCTION]
        else:
            cmd = ['bash', full_path]
```

Reply:
@chriscchien Why do we need to pass the function name from robot test code? What would be the drawback of letting the shell script handle it, like:
```
if [[ -n "${LONGHORN_STABLE_VERSION}" ]]; then
 install_longhorn_stable_version
else
  install_longhorn
fi
```

Reply:
@yangchiu, test case `Uninstallation Checks` also uses `longhorn-install.sh` to install Longhorn test version. If the script use environment variable to determin which version to install, this will result in `Uninstallation Checks` and `Pull backup created by another Longhorn system` can not running on the same environment.

Reply:
What about the following?
```
IS_INSTALL_STABLE_VERSION="${1:-false}"
if [[ "${IS_INSTALL_STABLE_VERSION}" == "true" ]]; then
  # do something
fi
```
Do you think it's more readable?

Reply:
Current code, run function name as first argument if the argument exist
```bash
./longhorn-install.sh install_longhorn_stable_version
```
suggestion (if I understand correctly)
```bash
./longhorn-install.sh true
```

Just think if enhance comment for more clear realize the scope as below is acceptable? if not, I will use the suggestion, thank you.
```bash
# Check if an argument is passed
if [[ $# -gt 0 ]]; then
    # Run the function specified in the first argument, 
    # ex: ./longhorn-install.sh install_longhorn_stable_version
    $1 
else
    # Run the default install_longhorn function if no argument is provided
    install_longhorn
fi
```

Reply:
How about the following:
```
IS_INSTALL_STABLE_VERSION=true ./longhorn-install.sh 
```
Can the shell script get the variable?

Reply:
Updated and tested.